linera-views/src/indexed_db.rs

```diff
@@ -0,0 +1,359 @@
+        Ok(Self::connect(config, "")
+            .await?
+            .database
+            .delete_object_store(namespace)?)
+    }
+}
+
+impl LocalKeyValueStore for IndexedDbStore {
+    type Error = IndexedDbContextError;
+}
+
+/// An implementation of [`crate::common::Context`] that stores all values in memory.
+pub type IndexedDbContext<E> = ContextFromStore<E, IndexedDbStore>;
+
+impl<E> IndexedDbContext<E> {
+    /// Creates a [`IndexedDbContext`].
+    pub fn new(store: IndexedDbStore, extra: E) -> Self {
+        Self {
+            store,
+            base_key: vec![],
+            extra,
+        }
+    }
+}
+
+/// Provides a `IndexedDbContext<()>` that can be used for tests.
+pub async fn create_indexed_db_test_context() -> IndexedDbContext<()> {
+    IndexedDbContext::new(
+        create_indexed_db_store_stream_queries(TEST_INDEX_DB_MAX_STREAM_QUERIES).await,
+        (),
+    )
+}
+
+/// Creates a test memory client for working.
+pub async fn create_indexed_db_store_stream_queries(max_stream_queries: usize) -> IndexedDbStore {
+    let common_config = CommonStoreConfig {
+        max_concurrent_queries: None,
+        max_stream_queries,
+        cache_size: 1000,
+    };
+    let config = IndexedDbStoreConfig { common_config };
+    let namespace = generate_test_namespace();
+    IndexedDbStore::connect(&config, &namespace).await.unwrap()
+}
+
+/// Creates a test memory store for working.
+#[cfg(with_testing)]
+pub async fn create_indexed_db_test_store() -> IndexedDbStore {
+    create_indexed_db_store_stream_queries(TEST_INDEX_DB_MAX_STREAM_QUERIES).await
+}
+
+/// The error type for [`IndexedDbContext`].
+#[derive(Error, Debug)]
+pub enum IndexedDbContextError {
+    /// Serialization error with BCS.
+    #[error("BCS error: {0}")]
+    Bcs(#[from] bcs::Error),
+
+    /// The value is too large for the IndexedDbStore
+    #[error("The value is too large for the IndexedDbStore")]
+    TooLargeValue,
+
+    /// The database is not consistent
+    #[error(transparent)]
+    DatabaseConsistencyError(#[from] DatabaseConsistencyError),
+
+    /// A DOM exception occurred in the IndexedDB operations
+    #[error("DOM exception: {}", self.to_string())]
+    Dom(web_sys::DomException),
+
+    /// JavaScript threw an exception whilst handling IndexedDB operations
+    #[error("JavaScript exception: {}", self.to_string())]
+    Js(wasm_bindgen::JsValue),
+}
+
+impl From<web_sys::DomException> for IndexedDbContextError {
+    fn from(dom_exception: web_sys::DomException) -> Self {
+        Self::Dom(dom_exception)
+    }
+}
+
+impl From<wasm_bindgen::JsValue> for IndexedDbContextError {
+    fn from(js_value: wasm_bindgen::JsValue) -> Self {
+        Self::Js(js_value)
+    }
+}
+
+impl From<IndexedDbContextError> for ViewError {
+    fn from(error: IndexedDbContextError) -> Self {
+        Self::ContextError {
+            backend: "memory".to_string(),
```

Comment:
I think "indexdb" is a fairer description of the origin of the error.

Reply:
Good catch, thanks!